yairl/scripts.py

```diff
@@ -148,77 +39,112 @@ def plot_fight_loss(env='CartPole-v1',
     disc_data = ([], [])
+
     def add_plot_disc(gen_mode=False):
         """
         gen_mode (bool): Whether the generator or the discriminator is active.
           We use this to color the data points.
         """
+        if n_plots_each_per_epoch <= 0:
+            return
+
         mode = "gen" if gen_mode else "dis"
         X, Y = gen_data if gen_mode else disc_data
         X.append(plot_idx)
         Y.append(trainer.eval_disc_loss())
         logging.info("plot idx ({}): {} disc loss: {}"
-                .format(mode, plot_idx, Y[-1]))
+                     .format(mode, plot_idx, Y[-1]))
+
     def show_plot_disc():
+        if n_plots_each_per_epoch <= 0:
+            return
+
         plt.scatter(disc_data[0], disc_data[1], c='g', alpha=0.7, s=4,
-                label="discriminator loss (dis step)")
+                    label="discriminator loss (dis step)")
         plt.scatter(gen_data[0], gen_data[1], c='r', alpha=0.7, s=4,
-                label="discriminator loss (gen step)")
+                    label="discriminator loss (gen step)")
         plt.title("epoch={}".format(epoch_num))
         plt.legend()
         _savefig_timestamp("plot_fight_loss_disc", interactive)
 
     gen_ep_reward = []
     rand_ep_reward = []
+
     def add_plot_gen():
-        env_vec = util.make_vec_env(env, 8)
+        if n_gen_plot_episodes <= 0:
+            return
+
         gen_policy = trainer.gen_policy
         rand_policy = util.make_blank_policy(env)
 
         gen_rew = util.rollout.total_reward(gen_policy, env,
-                n_episodes=n_gen_plot_episodes)/n_gen_plot_episodes
+                                            n_episodes=n_gen_plot_episodes) / n_gen_plot_episodes
         rand_rew = util.rollout.total_reward(rand_policy, env,
-                n_episodes=n_gen_plot_episodes)/n_gen_plot_episodes
+                                             n_episodes=n_gen_plot_episodes) / n_gen_plot_episodes
         gen_ep_reward.append(gen_rew)
         rand_ep_reward.append(rand_rew)
         logging.info("generator reward: {}".format(gen_rew))
         logging.info("random reward: {}".format(rand_rew))
+
     def show_plot_gen():
+        if n_gen_plot_episodes <= 0:
+            return
+
         plt.title("Cartpole performance (expert=500)")
         plt.xlabel("epochs")
         plt.ylabel("Average reward per episode (n={})"
-                .format(n_gen_plot_episodes))
+                   .format(n_gen_plot_episodes))
         plt.plot(gen_ep_reward, label="avg gen ep reward", c="red")
         plt.plot(rand_ep_reward, label="avg random ep reward", c="black")
         plt.legend()
         _savefig_timestamp("plot_fight_epreward_gen", interactive)
 
     add_plot_disc(False)
     add_plot_gen()
+
+    if n_plots_each_per_epoch <= 0:
+        n_gen_steps_per_plot = float('Inf')
+        n_disc_steps_per_plot = float('Inf')
+    else:
+        n_gen_steps_per_plot = int(n_gen_steps_per_epoch / n_plots_each_per_epoch + .5)
```

Comment:
Consider `int(math.ceil(...))`? Not sure if that's actually any cleaner

Reply:
I actually want to round though, not always round up.